### PR TITLE
Switch to JLD2.

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

File: docs/src/saving.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Then call the `save` method to save your variables:
 train.save(saver, session, "/path/to/variable_file")
 ```
 
-The newly-created "variable_file" is a [JLD](https://github.com/JuliaIO/JLD.jl) file that contains a mapping from variable names to their values. The value of variables can later be restored in a new Julia session with
+The newly-created "variable_file" is a [JLD2](https://github.com/simonster/JLD2.jl) file that contains a mapping from variable names to their values. The value of variables can later be restored in a new Julia session with
```
Comment (oxinabox) [inline docs/src/saving.md]: The variable file is still a JLD file.
It is just being processed and generated with JLD2.

**Comment by oxinabox** [comment]
This will be less  breaky avoiding #407 and similar

File: docs/src/saving.md (hunk 1 above)
Comment (malmaud) [reply docs/src/saving.md]: Oh, can JLD2 load a file saved with JLD1?

Comment (oxinabox) [reply docs/src/saving.md]: That was my understanding yes.

Comment (oxinabox) [reply docs/src/saving.md]: Ah apparently not

**Comment by oxinabox** [comment]
We should really have a test for this.

File: docs/src/saving.md (hunk 1 above)
Comment (oxinabox) [reply docs/src/saving.md]: Ignore me

**Comment by oxinabox** [comment]
Do we want to switch to BSON.jl instead?

**Comment by malmaud** [comment]
Oh ya, why not? It will probably be maintained better going forward, if nothing else. I did a quick check and they same to use the same amount of space for storing our files.

**Comment by oxinabox** [comment]
If this passes, lets merge it.
I just poked the CI.

**Pull request merged**